tools/matdump.c

```diff
@@ -539,7 +539,7 @@
         printf("%8" SIZE_T_FMTSTR, matvar->dims[0]);
         for ( i = 1; i < matvar->rank; i++ ) {
             if ( ceil(log10((double)matvar->dims[i])) + 1 < 32 )
-                cnt += sprintf(size + cnt, "x%" SIZE_T_FMTSTR, matvar->dims[i]);
+                cnt += snprintf(size + cnt, sizeof(size) - cnt, "x%" SIZE_T_FMTSTR, matvar->dims[i]);
```

Comment:
Funny how the bot is unhappy with a now-bounded call to snprintf, when previously sprintf was totally unbounded. :)

It's not entirely wrong though, as the return value is "the number of characters that would have been printed if the size were unlimited", so it theory it could return a large number, then underflow and wrap around.

But 32 bytes is big enough to print all 64 bit numbers in decimal.

Reply:
The isse is, that the return value cnt of s(n)printf is used as destination size + cnt in the next iteration. This can be dangerous. Are you willing to fix this on-the-fly?

Reply:
>Are you willing to fix this on-the-fly?

I'm willing to update this PR... but what is it you propose exactly?

>the return value cnt of s(n)printf is used as destination size + cnt in the next iteration

yes, this was the case before my patch too. it's a pretty idiomatic way to append to a string.

Looking at this code again, depending how many times the loop runs, the buffer may indeed be overwritten. So I'll at least update this thing to break out of the loop when the buffer is filled.